### ISSUE TITLE
I want to use seq2seq model for Q&A,but I don't know the output of the LSTM(decoder),is it a array of probability?? If it is an array of probability, is it necessary to present the output as one-hot representation?And then by finding the index of the max element to find the output word?

### ISSUE THREAD

**Issue by David931229** [fchollet/keras#6160]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [ ] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [ ] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by sallamander** [comment]
@David931229 The output of an LSTM isn't an array of probabilities; it's the result of some number of matrix multiplies and sums that depend upon the exact LSTM cell implementation ([Colah's blog here](http://colah.github.io/posts/2015-08-Understanding-LSTMs/) has a pretty good depiction of how LSTM cells work). The common way of translating that output to probabilities is by throwing a softmax over the output of the LSTM.  See the [babi_memnn.py](https://github.com/fchollet/keras/blob/master/examples/babi_memnn.py) or [babi_rnn.py](https://github.com/fchollet/keras/blob/master/examples/babi_rnn.py) examples for Q/A systems that do this.

**Comment by David931229** [comment]
@sallamander Thank you very much! You answer helps me a lot. And can you recommend me some examples for chatbot using keras seq2seq model? Thanks a lot!!!

**Comment by sallamander** [comment]
Hey @David931229.. I don't have any chatbot specific examples offhand, but any sequence to sequence learning examples will do, and/or generative models that produce text. The [Keras addition example](https://github.com/fchollet/keras/blob/master/examples/addition_rnn.py) and [Keras text generation example](https://github.com/fchollet/keras/blob/master/examples/lstm_text_generation.py) should be good places to get a jump start.

**Comment by David931229** [comment]
@sallamander Thank you very much! I will get a start from the two examples.Thanks again!

**Comment by philipperemy** [comment]
Close this issue if it's solved.
